Userland/Applications/CrashReporter/main.cpp

```diff
@@ -68,6 +79,8 @@ static TitleAndText build_backtrace(Coredump::Reader const& coredump, ELF::Core:
 
     auto timer = Core::ElapsedTimer::start_new();
     Coredump::Backtrace backtrace(coredump, thread_info, [&](size_t frame_index, size_t frame_count) {
+        if (!progress_window->window_id())
+            return;
```

Comment:
A more expressive check here would be `if (!progress_window->is_visible())`

That said, do we actually need this check if we exit the program on progress window close?

Reply:
I thought of some sort of race, where we're inside of the on_close function, but another thread tries to still push the update before we exit()'ed, or are both of those things in the same thread?

Reply:
We only set `m_window_id = 0` long after `on_close` has returned. So if your `on_close` calls `exit`, that never happens, and `if (!progress_window->window_id())` never triggers.

Also: Yes, most GUI things are on the EventLoop, so they can't execute in parallel.

Suggestion: Just remove it, `exit` shouldn't be able to have any race conditions with this.

Reply:
That does make sense, I removed it.